### PR TITLE
Improve docs of auth middleware and trigger service

### LINKED ISSUE #13048: Trigger Service: incorporate user management in the documentation

### LINKED ISSUE #13049: Auth Middleware: incorporate user management in the documentation

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
changelog_begin
changelog_end

Fix #13049
Fix #13048

Several improvements to the documentation for these two services. The main aim was
to add, where needed, references to the new user management service. Ultimately,
only the auth middleware really needed to add a note about it, since to the trigger
service this is (so far) completely transparent.

For the rest, I took advantage of the chance to polish the docs a bit and make a few
improvements (that I will describe more in depth in a self-review).

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/tools/trigger-service/index.rst [hunk 1]
```diff
@@ -14,120 +14,139 @@ Trigger Service
...
+
+        // Minimum idle connections for the database connection pool. Defaults to FIXME.
         min-idle = 4
-        //specifies the idle timeout for the database connection pool.
+
+        // Idle timeout for the database connection pool. Defaults to FIXME.
         idle-timeout = 12s
-        //specifies the connection timeout for database connection pool.
+
+        // Timeout for database connection pool. Defaults to FIXME.
```
Comment (stefanobaghino-da) [inline docs/source/tools/trigger-service/index.rst:105]: The defaults here need to be checked and/or fixed.

File: docs/source/tools/trigger-service/index.rst [hunk 2]
```diff
@@ -14,120 +14,139 @@ Trigger Service
...
       trigger-store {
+
+        // Mandatory. Database coordinates.
         user = "postgres"
         password = "password"
         driver = "org.postgresql.Driver"
         url = "jdbc:postgresql://localhost:5432/test?&ssl=true"
 
-        // prefix for table names to avoid collisions, empty by default
+        // Prefix for table names to avoid collisions. EXPERT ONLY. By default, this is empty and not used.
```
Comment (stefanobaghino-da) [inline docs/source/tools/trigger-service/index.rst:93]: Marked as "EXPERT ONLY" as it goes more or less together with `allow-existing-schema`.

File: docs/source/tools/trigger-service/index.rst [hunk 3]
```diff
@@ -14,120 +14,139 @@ Trigger Service
...
-
-The required config would look like
+The following snippet provides an example of what a possible `trigger-service.conf` configuration file could look like,
+alongside a few annotations with regards to the meaning of the configuration keys and possibly their default values.
 
 .. code-block:: none
 
     {
-      //dar file containing the trigger
+      // Mandatory. Paths to the DAR files containing the code executed by the trigger.
```
Comment (stefanobaghino-da) [inline docs/source/tools/trigger-service/index.rst:36]: I followed the following approach for edits here:
- document if a field is mandatory
- if it is not
  - if it doesn't have a default, mention it's optional
  - if it has a default, just mention it -- if something has a default, it must be optional
- one space after the comment token for descriptions
- no space after the comment token for values that have been commented out
- consistent capitalization

File: docs/source/tools/trigger-service/index.rst [hunk 4]
```diff
@@ -14,120 +14,139 @@ Trigger Service
...
       }
     }
 
-The above starts the Trigger Service using a number of default parameters. Most notably, the HTTP port the Trigger Service listens on which defaults to 8088.
-The above config file should list all available parameters, their defaults and descriptions.
-
-The trigger-service can also be started using cli-args as shown below, one can execute the command ``daml trigger-service --help`` to find all available parameters.
-
-.. note:: Configuration file is the recommended way to run trigger-service, running via cli-args is now deprecated
+The trigger service can also be started using command line arguments as shown below. The command ``daml trigger-service --help`` lists all available parameters.
```
Comment (stefanobaghino-da) [inline docs/source/tools/trigger-service/index.rst:139]: The fact that this is a list all available parameters, their defaults and descriptions has been moved above. The fact that command line parameters are now deprecated has been moved below.

File: docs/source/tools/trigger-service/index.rst [hunk 5]
```diff
@@ -137,15 +156,16 @@ Although as we'll see, the Trigger Service exposes an endpoint for end-users to
...
-~~~~~~~~~
+*********
 
 Start a trigger
-***************
+===============
 
-Start a trigger. In this example, Alice starts the trigger called ``trigger`` in a module called ``TestTrigger`` of a package with ID ``312094804c1468e2166bae3c9ba8b5cc0d285e31356304a2e9b0ac549df59d14``. The response contains an identifier for the running trigger that Alice can use in subsequent commands involving the trigger.
+Start a trigger. In this example, ``alice`` starts the trigger called ``trigger`` in a module called ``TestTrigger`` of a package with ID ``312094804c1468e2166bae3c9ba8b5cc0d285e31356304a2e9b0ac549df59d14``.
+The response contains an identifier for the running trigger that ``alice`` can use in subsequent commands involving the trigger.
```
Comment (stefanobaghino-da) [inline docs/source/tools/trigger-service/index.rst:165]: The party name did not match the one in the example.

File: docs/source/tools/trigger-service/index.rst [hunk 6]
```diff
@@ -183,20 +203,20 @@ HTTP Response
 
 
 Stop a trigger
-**************
+==============
 
-Stop a running trigger. Alice stops her running trigger like so.
+Stop a running trigger. In this example, the request asks to stop the trigger started above.
```
Comment (stefanobaghino-da) [inline docs/source/tools/trigger-service/index.rst:208]: More concrete reference.

File: docs/source/tools/trigger-service/index.rst [hunk 7]
```diff
@@ -14,120 +14,139 @@ Trigger Service
 
 The :ref:`running-a-no-op-trigger` section shows a simple method using the ``daml trigger`` command to arrange for the execution of a single trigger. Using this method, a dedicated process is launched to host the trigger.
 
-Complex workflows can require running many triggers for many parties and at a certain point, use of ``daml trigger`` with its process per trigger model becomes unwieldy. The Trigger Service provides the means to host multiple triggers for multiple parties running against a common ledger in a single process and provides a convenient interface for starting, stopping and monitoring them.
+Complex workflows can require running many triggers for many parties and at a certain point, use of ``daml trigger`` with its process-per-trigger model becomes unwieldy. The trigger service provides the means to host multiple triggers for multiple parties running against a common ledger in a single process and provides a convenient interface for starting, stopping and monitoring them.
```
Comment (S11001001) [inline docs/source/tools/trigger-service/index.rst:17]: I think "Trigger Service" is a proper name, but either way.

File: docs/source/tools/trigger-service/index.rst [hunk 8]
```diff
@@ -235,20 +255,19 @@ HTTP Response
...
 
 Status of a trigger
-*******************
+===================
 
-The status endoint returns metadata about the trigger like the
-party it is running as and the trigger id as well as the state the
-trigger is in (querying the acs, running, stopped).
+This endpoint returns data about a trigger, including the party on behalf of which it is running, its identifier,
+and its current state (querying the active contract set, running, or stopped).
```
Comment (stefanobaghino-da) [inline docs/source/tools/trigger-service/index.rst:261]: Metadata is data about data. Here we're just returning data.

File: docs/source/tools/trigger-service/index.rst [hunk 9]
```diff
@@ -14,120 +14,139 @@ Trigger Service
...
+
+        // Mandatory. Database coordinates.
         user = "postgres"
         password = "password"
         driver = "org.postgresql.Driver"
         url = "jdbc:postgresql://localhost:5432/test?&ssl=true"
 
-        // prefix for table names to avoid collisions, empty by default
+        // Prefix for table names to avoid collisions. EXPERT ONLY. By default, this is empty and not used.
         table-prefix = "foo"
```
Comment (S11001001) [inline docs/source/tools/trigger-service/index.rst:94]: For copypasta

```suggestion
        //table-prefix = "foo"
```

File: docs/source/tools/trigger-service/index.rst [hunk 10]
```diff
@@ -165,14 +185,14 @@ where
...
   ``${packageId}:${moduleName}:${identifierName}``. You can find the
   package id using ``daml damlc inspect path/to/trigger.dar | head -1``.
-- ``party`` is the party the trigger will be running as.
-- ``applicationId`` is an optional field to specify the application ID
+- ``party`` is the party on behalf of which the trigger is running.
+- ``applicationId`` is an optional field to specify the application id
   the trigger will use for command submissions. If omitted, the
   trigger will default to using its random UUID identifier returned in
-  the start request as the application ID.
+  the start request as the application id.
```
Comment (S11001001) [inline docs/source/tools/trigger-service/index.rst:192]: ego, superego? 😆

**Review by S11001001 — APPROVED** [review]

File: docs/source/tools/trigger-service/index.rst (hunk 7 above)
Comment (stefanobaghino-da) [reply docs/source/tools/trigger-service/index.rst:17]: I mostly wanted to make this consistent and the lower amount of changes was if I changed trigger service to be non-capitalized than to make auth middleware capitalized. But I'll change this to go the other way around.

File: docs/source/tools/trigger-service/index.rst (hunk 9 above)
Comment (stefanobaghino-da) [reply docs/source/tools/trigger-service/index.rst:94]: Good call, thanks.

File: docs/source/tools/trigger-service/index.rst (hunk 1 above)
Comment (stefanobaghino-da) [reply docs/source/tools/trigger-service/index.rst]: Addressed by https://github.com/digital-asset/daml/pull/13111/commits/9992ac7a1f947a40fc88278ea03d76af40439716

File: docs/source/tools/trigger-service/index.rst (hunk 7 above)
Comment (stefanobaghino-da) [reply docs/source/tools/trigger-service/index.rst]: Addressed by https://github.com/digital-asset/daml/pull/13111/commits/48a65754ee5e324cf2649d456c40ce990176c34e

File: docs/source/tools/trigger-service/index.rst (hunk 10 above)
Comment (stefanobaghino-da) [reply docs/source/tools/trigger-service/index.rst]: Addressed by https://github.com/digital-asset/daml/pull/13111/commits/49841ab2d667191071240eac8d6147670687587a

**Pull request merged**